crates/turbopack-ecmascript/src/references/mod.rs

```diff
@@ -318,52 +318,28 @@ pub(crate) async fn analyze_ecmascript_module(
-                                RequestVc::parse(Value::new(r.module_path.to_string().into())),
-                                Value::new(r.annotations.clone()),
-                                Some(ModulePartVc::new(ModulePart::ModuleEvaluation)),
-                            );
-                            import_references.push(vc);
-
-                            for symbol in symbols {
-                                let r = EsmAssetReferenceVc::new(
-                                    origin,
-                                    RequestVc::parse(Value::new(r.module_path.to_string().into())),
-                                    Value::new(r.annotations.clone()),
-                                    Some(ModulePartVc::new(ModulePart::Export(StringVc::cell(
-                                        symbol.to_string(),
-                                    )))),
-                                );
-                                import_references.push(r);
+            for r in eval_context.imports.references() {
+                let r = EsmAssetReferenceVc::new(
+                    origin,
+                    RequestVc::parse(Value::new(r.module_path.to_string().into())),
+                    Value::new(r.annotations.clone()),
+                    if options.import_parts {
+                        match &r.imported_symbol {
+                            ImportedSymbol::ModuleEvaluation => {
+                                Some(ModulePartVc::module_evaluation())
                             }
+                            ImportedSymbol::Symbol(name) => {
+                                Some(ModulePartVc::export(name.to_string()))
+                            }
+                            ImportedSymbol::Namespace => None,
                         }
-                        ImportedSymbols::Namespace => {
-                            let r = EsmAssetReferenceVc::new(
-                                origin,
-                                RequestVc::parse(Value::new(r.module_path.to_string().into())),
-                                Value::new(r.annotations.clone()),
-                                None,
-                            );
-                            import_references.push(r);
-                        }
-                    }
-                }
-            } else {
-                for r in eval_context.imports.references() {
-                    let vc = EsmAssetReferenceVc::new(
-                        origin,
-                        RequestVc::parse(Value::new(r.module_path.to_string().into())),
-                        Value::new(r.annotations.clone()),
-                        None,
-                    );
-                    import_references.push(vc);
-                }
+                    } else {
+                        None
+                    },
+                );
+                import_references.push(r);
             }
+
             for r in import_references.iter_mut() {
```

Comment:
```suggestion
            // Avoid adding duplicate references to the analysis
            // (this can work on unresolved Vcs as we can deduplicate on operations)
            for r in IndexSet::from(import_references.iter()) {
```

Avoiding duplicate references